app/assets/javascripts/watchlist.js

```diff
@@ -0,0 +1,431 @@
+        break;
+      default:
+        console.log(`${condition} is not valid`);
+        return;
+    }
+    conditions_html += `
+        <div class="ui circular label condition" data-content="${violation_string}" data-variation="wide"> 
+          ${condition_string}
+        </div>`
+  });
+  return conditions_html;
+}
+
+function get_buttons_html(user_id, tab, archived_instances) {
+  var archived_icon = (user_id in archived_instances) ? 
+      `<div class="left ui icon" data-content="Student also appears in archived">
+        <i class="exclamation circle icon"></i>
+      </div>` : "";
+
+  switch(tab) {
+    case "new":
+      return `
+        <div class="students-buttons-right"> 
+          ${archived_icon}
+          <button class="ui submit tiny button contact_single"><i class="mail outline icon"></i>CONTACT</button>
+          <button class="ui submit tiny button resolve_single"><i class="check circle icon"></i>RESOLVE</button>
+        </div>`
+    case "contacted":
+      return `
+        <div class="students-buttons-right"> 
+          ${archived_icon}
+          <button class="ui submit tiny button resolve_single"><i class="check circle icon"></i>RESOLVE</button>
+        </div>`
+    case "resolved":
+      return `
+        <div class="students-buttons-right"> 
+          ${archived_icon}
+        </div>`
+    case "archived":
+      return "";
+    default:
+      console.log(`${tab} is not a valid tab`);
+      return;
+  }
+}
+
+function get_row_html(user_id, instance, tab, archived_instances) {
+  var name = instance["name"];
+  var condition_types = instance["conditions"];
+  var course_id = instance["course_id"];
+
+  var name_html = get_name_html(name);
+  var gradebook_link_html = get_gradebook_link_html(course_id, user_id);
+  var conditions_html = get_condition_html(condition_types);
+  var buttons_html = get_buttons_html(user_id, tab, archived_instances);
+  return `
+      <div class="ui segment" id=${user_id}>
+        ${name_html}
+        ${gradebook_link_html}
+        ${conditions_html}
+        ${buttons_html}
+      </div>`;
+}
+
+function addInstanceToDict(instancesDict, id, user_id, course_id, user_name, user_email, condition_type, violation_info) {
+  if (user_id in instancesDict) {
+    instancesDict[user_id]["conditions"][condition_type] = violation_info;
+    instancesDict[user_id]["instance_ids"].push(id);
+  } else {
+    instancesDict[user_id] = {
+      "name": user_name, 
+      "email": user_email,
+      "course_id": course_id,
+      "conditions": {},
+      "instance_ids": [id]
+    };
+    instancesDict[user_id]["conditions"][condition_type] = violation_info;
+  }
+}
+
+function get_watchlist_function(){
+  var new_instances = {}
+  var contacted_instances = {}
+  var resolved_instances = {}
+  var archived_instances = {}
+
+  var selected_user_ids = []
+
+	$.getJSON(watchlist_endpoints['get'],function(data, status){
+	    if(status=='success') {
```

Comment:
there should preferably be an error situation as well

Reply:
Made the error case show the same banner as when the POST requests don't work.